### PR TITLE
add "chain" example

### PR DESCRIPTION
**Pull request by beiwei30**

### CODE REVIEW

File: chain/backend/pkg/dog_service.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pkg
+
+import (
+	"fmt"
+	"github.com/apache/dubbo-go/config"
```
Comment (AlexStocks) [inline chain/backend/pkg/dog_service.go:22]: pls split this import package.

File: chain/backend/pkg/lion_service.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pkg
+
+import (
+	"fmt"
+	"github.com/apache/dubbo-go/config"
```
Comment (AlexStocks) [inline chain/backend/pkg/lion_service.go:22]: pls split this import block.

File: chain/backend/pkg/tiger_service.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pkg
+
+import (
+	"fmt"
+	"github.com/apache/dubbo-go/config"
```
Comment (AlexStocks) [inline chain/backend/pkg/tiger_service.go:22]: split it, pls.

File: chain/backend/tests/integration/main_test.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	_ "github.com/apache/dubbo-go/cluster/loadbalance"
+	_ "github.com/apache/dubbo-go/common/proxy/proxy_factory"
+	_ "github.com/apache/dubbo-go/filter/filter_impl"
+	_ "github.com/apache/dubbo-go/metadata/service/inmemory"
+	_ "github.com/apache/dubbo-go/protocol/dubbo"
+	_ "github.com/apache/dubbo-go/registry/protocol"
+	_ "github.com/apache/dubbo-go/registry/zookeeper"
+)
+
+import (
```
Comment (AlexStocks) [inline chain/backend/tests/integration/main_test.go:35]: move this import block to the the top, pls.

File: chain/frontend/cmd/client.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
```
Comment (AlexStocks) [inline chain/frontend/cmd/client.go:20]: pls split this import block.

File: chain/middle/cmd/server.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+import (
+	_ "github.com/apache/dubbo-go-samples/chain/middle/pkg"
```
Comment (AlexStocks) [inline chain/middle/cmd/server.go:29]: move this dir to the 2rd import block, pls.

File: chain/middle/pkg/american_service.go [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pkg
+
+import (
```
Comment (AlexStocks) [inline chain/middle/pkg/american_service.go:20]: split this import block, pls.

File: chain/middle/pkg/chinese_service.go [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pkg
+
+import (
```
Comment (AlexStocks) [inline chain/middle/pkg/chinese_service.go:20]: split it, pls.

File: chain/middle/tests/integration/main_test.go [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+)
+
+import (
+	"os"
+	"testing"
+	"time"
+)
+
+var chinese = new(ChineseService)
+var american = new(AmericanService)
```
Comment (AlexStocks) [inline chain/middle/tests/integration/main_test.go:42]: merge this two vars to the same var block as the following.

```go
 var (
  chinese = xxx
  american = xxx
)

File: chain/middle/tests/integration/services_test.go [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package integration
+
+import (
```
Comment (AlexStocks) [inline chain/middle/tests/integration/services_test.go:22]: split it, pls.

File: .run/chain-backend-server.run.xml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<component name="ProjectRunConfigurationManager">
```
Comment (zouyx) [inline .run/chain-backend-server.run.xml:1]: what does this file use for?

**Review by zouyx — APPROVED** [review]

File: chain/backend/pkg/dog_service.go (hunk 1 above)
Comment (beiwei30) [reply chain/backend/pkg/dog_service.go:25]: done

File: chain/backend/pkg/lion_service.go (hunk 2 above)
Comment (beiwei30) [reply chain/backend/pkg/lion_service.go:25]: done.

File: chain/backend/tests/integration/main_test.go (hunk 4 above)
Comment (beiwei30) [reply chain/backend/tests/integration/main_test.go:34]: done.

File: chain/middle/cmd/server.go (hunk 6 above)
Comment (beiwei30) [reply chain/middle/cmd/server.go:29]: done.

File: chain/middle/pkg/american_service.go (hunk 7 above)
Comment (beiwei30) [reply chain/middle/pkg/american_service.go:20]: dond.

File: chain/middle/pkg/chinese_service.go (hunk 8 above)
Comment (beiwei30) [reply chain/middle/pkg/chinese_service.go:20]: done.

File: chain/middle/tests/integration/main_test.go (hunk 9 above)
Comment (beiwei30) [reply chain/middle/tests/integration/main_test.go:41]: done.

File: .run/chain-backend-server.run.xml (hunk 11 above)
Comment (beiwei30) [reply .run/chain-backend-server.run.xml:1]: pre-configured 'run configuration' in GoLand

File: chain/backend/pkg/tiger_service.go (hunk 3 above)
Comment (beiwei30) [reply chain/backend/pkg/tiger_service.go:25]: done.

File: chain/frontend/cmd/client.go (hunk 5 above)
Comment (beiwei30) [reply chain/frontend/cmd/client.go:20]: done.

**Review by AlexStocks — APPROVED** [review]

**Review by LaurenceLiZhixin — APPROVED** [review]

**Pull request merged**